packages/esm-appointments-app/src/appointments/common-components/appointments-actions.test.tsx

```diff
@@ -66,21 +73,43 @@ describe('AppointmentActions', () => {
     mutate: () => {},
   };
 
-  beforeAll(() => {
-    jest.useFakeTimers();
-
-    const currentDateTime = new Date();
-    currentDateTime.setHours(12);
-    currentDateTime.setMinutes(0);
+  afterAll(() => {
+    jest.useRealTimers();
+  });
 
-    jest.setSystemTime(currentDateTime);
+  it('renders the check in button when appointment is today and the patient has not checked in and check in button enabled', () => {
+    useConfig.mockImplementation(() => ({
+      checkInButton: { enabled: true },
+      checkOutButton: { enabled: true },
+    }));
+    useTodaysVisits.mockImplementation(() => ({
+      visits: [],
+    }));
+    const props = { ...defaultProps };
+    const { getByText } = render(<AppointmentActions {...props} />);
+    const button = getByText('Check In');
+    expect(button).toBeInTheDocument();
   });
 
-  afterAll(() => {
-    jest.useRealTimers();
+  it('does not renders the check in button when appointment is today and the patient has not checked in but the check-in button is disabled', () => {
+    useConfig.mockImplementation(() => ({
+      checkInButton: { enabled: false },
+      checkOutButton: { enabled: true },
+    }));
+    useTodaysVisits.mockImplementation(() => ({
+      visits: [],
+    }));
+    const props = { ...defaultProps };
+    const { queryByText } = render(<AppointmentActions {...props} />);
+    const button = queryByText('Check In');
+    expect(button).not.toBeInTheDocument();
   });
 
-  it('renders the correct button when the patient has checked out', () => {
+  it('renders the checked out button when the patient has checked out', () => {
```

Comment:
shouldn't be a button at this point, right?

Reply:
One might thing that, but it's a non-clickable button, maybe for styling reasons?